library/cmake/Modules/FindPHP.cmake

```diff
@@ -31,10 +31,24 @@
 #=============================================================================
 # (To distribute this file outside of CMake, substitute the full
 #  License text for the above reference.)
+#
+# This file has been modified to include the option to specify a "Development" component
+# and find Zend headers.  © Nightwave Studios, 2017.
+#
+
+set(PHP_FOUND FALSE)
+set(PHP_DEVELOPMENT_COMPONENT_FOUND FALSE)
+
+if(PHP_FIND_COMPONENTS)
+	foreach(component ${PHP_FIND_COMPONENTS})
+		string(TOUPPER ${component} _COMPONENT)
+		set(PHP_USE_${_COMPONENT} 1)
+	endforeach()
+endif()
```

Comment:
I really don't get why this part is necessary :). Is it for normalizing variable names? Especially `set(PHP_USE_${_COMPONENT} 1)` is confusing me.

Reply:
It is, yes, I copied and pasted it from CMake's `FindQt4.cmake` module.  It's designed to handle multiple components, but we can shorten this down to only accept one if you would prefer?

Reply:
> but we can shorten this down to only accept one if you would prefer?

I don't mind handling several components. In particular with a view to your wish to add this module to your personal project, it might be useful to have a "generic" component handling. However, I would suggest to put this snippet next to the actual component handling.

Reply:
I'm confused by what you mean by a generic component.  Can you possibly explain more?